### PR TITLE
dumpling: Set default driverCfg to use TLS first

### PR DESCRIPTION
**Pull request by Daemonxiao**
### What problem does this PR solve?

Use TLS  connection by default in the dumpling.

### What is changed and how it works?

1. Add `AllowFallbackToPlaintext = true` in `driverCfg`.
2. Set `MinVersion: tls.VersionTLS10`

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: dumpling/export/config.go [hunk 1]
```diff
@@ -221,6 +222,13 @@ func (conf *Config) GetDriverConfig(db string) *mysql.Config {
 	driverCfg.MaxAllowedPacket = 0
 	if conf.Security.DriveTLSName != "" {
```
Comment (lance6716) [inline dumpling/export/config.go:223]: Seems we also don't need it now, originally we use `DriveTLSName` for `RegisterTLSConfig`. Now we can remove `registerTLSConfig` and change `DriveTLSName` to `TLSConfig` to let `driverCfg.TLS` use it.

And can you also do this adjustment for lightning in next PR?

File: cmd/explaintest/r/explain_complex.result [hunk 2]
```diff
@@ -249,7 +249,7 @@ Sort	1.00	root		test.org_department.left_value
...
   └─Selection	0.01	root		or(eq(test.org_employee_position.user_id, 62), or(eq(test.org_department.id, 20), eq(test.org_department.id, 20)))
     └─HashJoin	0.02	root		left outer join, equal:[eq(test.org_position.id, test.org_employee_position.position_id)]
-      ├─IndexJoin(Build)	0.01	root		left outer join, inner:IndexLookUp, outer key:test.org_department.id, inner key:test.org_position.department_id, equal cond:eq(test.org_department.id, test.org_position.department_id)
+      ├─IndexHashJoin(Build)	0.01	root		left outer join, inner:IndexLookUp, outer key:test.org_department.id, inner key:test.org_position.department_id, equal cond:eq(test.org_department.id, test.org_position.department_id)
```
Comment (lance6716) [inline cmd/explaintest/r/explain_complex.result:252]: this is unexpected 🤔

Comment (Daemonxiao) [reply cmd/explaintest/r/explain_complex.result:252]: umm... I will correct it according to the result of the check_dev action.

**Review by lance6716 — APPROVED** [review]

**Review by lichunzhu — APPROVED** [review]
LGTM

**Pull request merged**